File: JsonToPdf/common/schema.go
package common

type JsonSchema struct {
	CvBaseInfo CvBaseInfo  `json:"cv_base_info"`
	CvEduInfo  []CvEduInfo `json:"cv_edu_info"`
	CvResume   []CvResume  `json:"cv_resume"`
	CvSkills   CvSkills    `json:"cv_skills"`
}

type CvBaseInfo struct {
	Name      string `json:"name"`
	Birthday  int64  `json:"birthday"`
	Mobile    int64  `json:"mobile"`
	Email     string `json:"email"`
	PhotoPath string `json:"photo_path"`
} // CvBaseInfo

type CvEduInfo struct {
	School         string `json:"school"`
	Major          string `json:"major"`
	AcademicDegree string `json:"academic_degree"`
	StartTime      int64  `json:"start_time"`
	EndTime        int64  `json:"end_time"`
} // CvEduInfo

type CvResume struct {
	Company     string        `json:"company"`
	ResumeSteps []ResumeSteps `json:"resume_steps"`
	Achievement string        `json:"achievement"`
} // CvResume

type ResumeSteps struct {
	Position         string `json:"position"`
	Department       string `json:"department"`
	StartTime        int64  `json:"start_time"`
	EndTime          int64  `json:"end_time"`
	Responsibilities string `json:"responsibilities"`
} // ResumeSteps

type CvSkills []string // Skills

File: src-tauri/src/cv_kawa_main/schema.rs
use serde::{Deserialize, Serialize};

#[derive(Default, Debug, Serialize, Deserialize, Clone)]
pub struct CV {
    pub cv_base_info: CVBaseInfo,
    pub cv_edu_info: Vec<CVEduInfo>,
    pub cv_resume: Vec<CVResume>,
    pub cv_skills: Vec<String>,
}

#[derive(Default,Debug, Serialize, Deserialize,Clone)]
pub struct CVResume {
    pub company: String,
    pub resume_steps: Vec<CVResumeStep>,
    pub achievement: String,
}

#[derive(Default,Debug, Serialize, Deserialize,Clone)]
pub struct CVResumeStep{
    pub position: String,
    pub department: String,
    pub start_time: i64,
    pub end_time: i64,
    pub responsibilities: String,
}

#[derive(Default,Debug, Serialize, Deserialize,Clone)]
pub struct CVEduInfo {
    pub school: String,
    pub major: String,
    pub academic_degree: String,
    pub start_time: i64,
    pub end_time: i64,
}

#[derive(Default,Debug, Serialize, Deserialize,Clone)]
pub struct CVBaseInfo {
    pub name: String,
    pub birthday: i64,
    pub mobile: i64,
    pub email: String,
    pub photo_path: Option<String>,
}
